packages/server/src/integration-test/postgres.spec.ts

```diff
@@ -0,0 +1,663 @@
+    expect(res.body).toEqual({
+      config: {
+        ca: false,
+        database: "postgres",
+        host,
+        password: "--secret-value--",
+        port,
+        rejectUnauthorized: false,
+        schema: "public",
+        ssl: false,
+        user: "postgres",
+      },
+      plus: true,
+      source: "POSTGRES",
+      type: "datasource",
+      _id: expect.any(String),
+      _rev: expect.any(String),
+      createdAt: expect.any(String),
+      updatedAt: expect.any(String),
+      entities: expect.any(Object),
+    })
+  })
+
+  describe("create a row", () => {
+    const createRow = (tableId: string | undefined, body: object) =>
+      makeRequest("post", `/api/${tableId}/rows`, body)
+
+    test("Given than no row exists, adding a new row persists it", async () => {
+      const newRow = makeRandomRow()
+
+      const res = await createRow(postgresTable._id, newRow)
+
+      expect(res.status).toBe(200)
+
+      const persistedRows = await config.getRows(postgresTable._id!)
+      expect(persistedRows).toHaveLength(1)
+
+      const expected = {
+        ...res.body,
+        ...newRow,
+      }
+
+      expect(persistedRows).toEqual([expect.objectContaining(expected)])
+    })
+
+    test("Given than no row exists, multiple rows can be persisted", async () => {
+      const numberOfRows = 10
+      const newRows = Array(numberOfRows).fill(makeRandomRow())
+
+      for (const newRow of newRows) {
+        const res = await createRow(postgresTable._id, newRow)
+        expect(res.status).toBe(200)
+      }
+
+      const persistedRows = await config.getRows(postgresTable._id!)
+      expect(persistedRows).toHaveLength(numberOfRows)
+      expect(persistedRows).toEqual(
+        expect.arrayContaining(newRows.map(expect.objectContaining))
+      )
+    })
+  })
+
+  describe("update a row", () => {
+    const updateRow = (tableId: string | undefined, body: Row) =>
+      makeRequest("patch", `/api/${tableId}/rows`, body)
+
+    test("Given than a row exists, updating it persists it", async () => {
+      let { row } = _.sample(await populateRows(10))!
+
+      const newName = generator.name()
+      const newValue = generator.age()
+      const updatedRow = {
+        ...row,
+        name: newName,
+        value: newValue,
+      }
+
+      const res = await updateRow(postgresTable._id, updatedRow)
+
+      expect(res.status).toBe(200)
+      expect(res.body).toEqual(updatedRow)
+
+      const persistedRow = await config.getRow(postgresTable._id!, row.id)
+
+      expect(persistedRow).toEqual(
+        expect.objectContaining({
+          id: row.id,
+          name: newName,
+          value: newValue,
+        })
+      )
+    })
+  })
+
+  describe("delete a row", () => {
+    const deleteRow = (
+      tableId: string | undefined,
+      body: Row | { rows: Row[] }
+    ) => makeRequest("delete", `/api/${tableId}/rows`, body)
+
+    test("Given than a row exists, delete request removes it", async () => {
```

Comment:
I find breaking up the `Given` statements into their own `describe` blocks can help with producing re-usable scenarios / highlight negative and error test cases. e.g.

```ts
describe("delete a row", () => {
    it("produces 404", () => {
    })

    describe("rows exists", async () => {
      
      let rows: Row[]
      
      beforeEach(async () => {
         rows = await populateRows(numberOfInitialRows)
      })
        
      it("deletes a row", () => {})
      it("deletes multiple rows", () => {})
    })
})
```